smpp/server.go

```diff
@@ -178,19 +185,20 @@ func (srv *server) Serve() {
 // handle new clients.
 func (srv *server) handle(c *conn) {
 	defer c.Close()
-	if err := srv.auth(c); err != nil {
-		if err != io.EOF {
-			log.Println("Server auth failed:", err)
-		}
-		return
-	}
+
 	// Use connSwitch to have synced read/write
 	s := &session{conn: &connSwitch{}}
 	s.conn.Set(c)
 	s.id = randomString(IDLen)
 	srv.mu.Lock()
 	srv.s[s.id] = s
 	srv.mu.Unlock()
+
+	if err := srv.auth(c, s); err != nil {
+		log.Println("Server auth failed:", err)
+		return
```

Comment:
The logic is in `auth` and it depends. If there's a registered Auth Handler, then it leaves everything to it (verify creds, write response, catch errors). If there is not, it will close the session without writing a response back, which has been the behavior since I forked this repo. But now that you mention it, I think I should add also the response writing there.